src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+        """
+        cost_sum = 0
+        try:
+            cost_run, sexp = block_program.run_with_cost([])
+            cost_sum += cost_run
+        except EvalError:
+            return False
+
+        for name_solution in sexp.as_iter():
+            _ = name_solution.as_python()
+            if len(_) != 2:
+                return False
+            if not isinstance(_[0], bytes) or len(_[0]) != 32:
+                return False
+            coin_name = bytes32(_[0])
+            if not isinstance(_[1], list) or len(_[1]) != 2:
+                return False
+            puzzle_solution_program = name_solution.rest().first()
+            puzzle_program = puzzle_solution_program.first()
+            try:
+                error, conditions_dict, cost_run = conditions_dict_for_solution(
+                    puzzle_solution_program
+                )
+                cost_sum += cost_run
+                if error:
+                    return False
+            except clvm.EvalError:
+
+                return False
+            if conditions_dict is None:
+                conditions_dict = {}
+
+            if ConditionOpcode.CREATE_COIN in conditions_dict:
+                created_output_conditions = conditions_dict[ConditionOpcode.CREATE_COIN]
+            else:
+                continue
+            for cvp in created_output_conditions:
+                result = await self.wallet_state_manager.puzzle_store.wallet_info_for_puzzle_hash(
+                    cvp.var1
+                )
+                if result is None:
+                    continue
+
+                wallet_id, wallet_type = result
+                if wallet_id != self.wallet_info.id:
+                    continue
+
+                coin = None
+                for removed in removals:
+                    if removed.name() == coin_name:
+                        coin = removed
+                        break
+
+                if coin is not None:
+                    if did_wallet_puzzles.check_is_did_puzzle(puzzle_program):
+                        inner_puzzle_hash = (
+                            did_wallet_puzzles.get_innerpuzzle_from_puzzle(
+                                puzzle_program
+                            )
+                        )
+                        self.log.info(
+                            f"parent: {coin_name} inner_puzzle for parent is {inner_puzzle_hash.hex()}"
+                        )
+
+                        await self.add_parent(
+                            coin_name,
+                            CCParent(
+                                coin.parent_coin_info, inner_puzzle_hash, coin.amount
+                            ),
+                        )
+
+                return True
+
+        return False
+
+    async def generator_received(
+        self, height: uint32, header_hash: bytes32, generator: Program, action_id: int
+    ):
+        """ Notification that wallet has received a generator it asked for. """
```

Comment:
I am not sure I get it... are you saying that this is a callback called by some other code?